### PR TITLE
test: make QEMU tests work on Semaphore

### PR DESCRIPTION
**Pull request by zonque**

### CODE REVIEW

**Comment by zonque** [comment]
Don't merge this one just yet, I merely need the branch to run some more tests ...

File: test/TEST-01-BASIC/test.sh [hunk 1]
```diff
@@ -57,6 +57,13 @@ ExecStart=/bin/sh -x -c 'systemctl --failed --no-legend --no-pager > /failed ; e
 Type=oneshot
 EOF
 
+        # mask some services that we do not want to run in these tests
```
Comment (evverx) [inline test/TEST-01-BASIC/test.sh]: @zonque , you mask these services in `nspawn` tests too.
Maybe,

```
setup_testsuite
setup_nspawn_root

ln -s /dev/null $initdir/etc/systemd/system/systemd-hwdb-update.service
...
```

?

**Comment by evverx** [comment]
@zonque , I fixed

```
ls: cannot access /var/tmp/systemd-test.R60qNU/journal/*/*.journal: No such file or directory
W: can't run systemd-nspawn, skipping
TEST RUN: Basic systemd setup [OK]
make: Leaving directory `/home/runner/systemd/test/TEST-01-BASIC'
```

see https://github.com/systemd/systemd/pull/1960

File: test/TEST-01-BASIC/test.sh (hunk 1 above)
Comment (zonque) [inline test/TEST-01-BASIC/test.sh]: @evverx, jup, you're right. Updated accordingly.

My tests on Semaphore succeed now, btw. So I think this is safe to merge.

**Comment by evverx** [comment]
@zonque , I fixed initrd searching on Debian/Ubuntu. See https://github.com/systemd/systemd/pull/1966

**Comment by zonque** [comment]
@evverx, the machined detection now works for you on Ubuntu 15.10?

**Comment by evverx** [comment]
@zonque , yes, it works. Thanks!

```
$ sudo make clean setup run
...
Spawning container nspawn-root on /var/tmp/systemd-test.gSYvFY/nspawn-root.
...

$ sudo apt-get remove systemd-container
$ sudo make clean setup run
...
W: can't run systemd-nspawn, skipping
...
```

**Comment by poettering** [comment]
Merged, would still be curious about my questions though.

**Pull request merged**